helm: Add hubble section

michi-covalent opened the pull request:
2 commits:

- add a hidden `cilium observe` subcommand to get flows from hubble.
- update helm.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/10358)
<!-- Reviewable:end -->

michi-covalent committed:
helm: Add hubble section

Add hubble section to values.yaml, and add new Hubble-related fields
to the configmap. The configmap has these new hubble-related fields:

- hubble-listen-addresses
    List of addresses for Hubble to listen to. Disabled if empty.
- hubble-flow-buffer-size
    Number of recent flows for Hubble to cache. Defaults to 4096.
- hubble-event-queue-size
    Buffer size of the events channel. Defaults to 128.
- hubble-metrics-server
    Address for the metric server to listen to. Disabled if empty.
- hubble-metrics
    List of metrics to collect.

This PR also adds 2 environment variables if Hubble is enabled:

- HUBBLE_GROUP_NAME
     Group for Hubble's unix domain sockets. Hardcoded to be `cilium`.
- HUBBLE_DEFAULT_SOCKET_PATH
     Default Hubble gRPC endpoint for observe/status commands. This
     is set to the first address in hubble-listen-addresses.

Here is a sample helm command to configure Hubble-related settings:

    helm template cilium \
      --set global.hubble.listenAddresses="{unix:///var/run/cilium/hubble.sock,unix:///var/run/cilium/hubble.sock2}" \
      --set global.hubble.eventQueueSize=1234 \
      --set global.hubble.flowBufferSize=5678 \
      --set global.hubble.metricsServer=":7071" \
      --set global.hubble.metrics="{dns:query;ignoreAAAA,drop,tcp,flow,port-distribution,icmp,http}"

Ref #9925

Signed-off-by: Michi Mutsuzaki <michi@isovalent.com>

aanm reviewed (changes requested):
I don't see where `HUBBLE_DEFAULT_SOCKET_PATH` nor `HUBBLE_GROUP_NAME` are used in the cilium's code base so I assume this is waiting for #10238 to be merged first?

michi-covalent committed:
cli: Add a hidden 'observe' subcommand

The observe subcommand executes Hubble's observe command. We are still
trying to figure out how to distribute Hubble's CLI. This is a short
term workaround so that people can use it to interact with Hubble in
the meantime.

Ref #9925

Signed-off-by: Michi Mutsuzaki <michi@isovalent.com>

gandro commented on install/kubernetes/cilium/values.yaml:
```diff
@@ -359,3 +359,42 @@ global:
+    # Hubble is disabled if the list is empty.
+    listenAddresses: []
+    # Buffer size of the channel Hubble uses to receive monitor events. Defaults to 128.
+    eventQueueSize: ~
```
Apparently I missed this in the other PR, but 128 (which is taken from stand-alone) is way to too small for embedded mode, Hubble will drop most flows with this setting (https://github.com/cilium/hubble/issues/161).

This means we need to the bump the default for `option.HubbleEventQueueSize` here as well:
https://github.com/cilium/cilium/blob/8d28ab13be7d2aeca75702b89445a0d26c4815ca/daemon/daemon_main.go#L738

I propose we use the same value as we use for `MonitorQueueSize`:
https://github.com/cilium/cilium/blob/8d28ab13be7d2aeca75702b89445a0d26c4815ca/pkg/option/config.go#L2166-L2171

tgraf replied:
Let's fix it in a separate PR so it's documented properly why we bumped it.

michi-covalent replied:
https://github.com/cilium/cilium/pull/10488 cc @gandro

gandro reviewed (commented):
See comment regarding event queue size. I'm fine if we fix this in a follow-up PR though.
